### PR TITLE
New extension hook entry_auto_read

### PR DESCRIPTION
**Pull request by Alkarex**
For extensions to be notified of articles being automatically marked as read for various reasons

### CODE REVIEW

File: lib/Minz/ExtensionManager.php [hunk 1]
```diff
@@ -26,6 +26,10 @@ final class Minz_ExtensionManager {
 			'list' => array(),
 			'signature' => 'OneToOne',
 		),
+		'entry_auto_read' => array(	// function(FreshRSS_Entry $entry, string $why) -> none
+			'list' => array(),
+			'signature' => 'PassArguments',
+		),
```
Comment (ColonelMoutarde) [inline lib/Minz/ExtensionManager.php]: use [] for arrays ;)

Comment (Alkarex) [reply lib/Minz/ExtensionManager.php]: Another time, not to mess the rest of the PR here

Comment (Alkarex) [reply lib/Minz/ExtensionManager.php]: https://github.com/FreshRSS/FreshRSS/pull/5504

File: app/Controllers/feedController.php [hunk 2]
```diff
@@ -479,8 +479,16 @@ public static function actualizeFeed(int $feed_id, string $feed_url, bool $force
...
 							$entry->_isFavorite(null);	// Do not change favourite state
+							$entry->_isRead($mark_updated_article_unread ? false : null);	//Change is_read according to policy.
+							if ($mark_updated_article_unread) {
+								Minz_ExtensionManager::callHook('entry_auto_unread', $entry, 'updated_article');
+							}
+
+							$entry->applyFilterActions($titlesAsRead);
+							if ($readWhenSameTitleInFeed > 0) {
+								$titlesAsRead[$entry->title()] = true;
+							}
```
Comment (Alkarex) [inline app/Controllers/feedController.php:491]: This was wrong. Reverted in https://github.com/FreshRSS/FreshRSS/pull/5561

**Pull request merged**